src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -52,69 +52,427 @@ public class RepoConfiguration {
-            boolean isFindingPreviousAuthorsPerformed, boolean isIgnoredFileAnalysisSkipped,
-            List<String> ignoredAuthorsList, boolean isIgnoredAuthorsListOverriding, String extraOutputFolderName) {
-        this.authorConfig = new AuthorConfiguration(location, branch);
-        this.location = location;
-        this.branch = location.isEmpty() ? DEFAULT_BRANCH : branch;
-        this.ignoreGlobList = ignoreGlobList;
-        this.fileSizeLimit = fileSizeLimit;
-        this.isStandaloneConfigIgnored = isStandaloneConfigIgnored;
-        this.isFileSizeLimitIgnored = isFileSizeLimitIgnored;
-        this.fileTypeManager = new FileTypeManager(formats);
-        this.ignoreCommitList = ignoreCommitList;
-        this.isFormatsOverriding = isFormatsOverriding;
-        this.isIgnoreGlobListOverriding = isIgnoreGlobListOverriding;
-        this.isIgnoreCommitListOverriding = isIgnoreCommitListOverriding;
-        this.isFileSizeLimitOverriding = isFileSizeLimitOverriding;
-        this.isShallowCloningPerformed = isShallowCloningPerformed;
-        this.isFindingPreviousAuthorsPerformed = isFindingPreviousAuthorsPerformed;
-        this.isIgnoredFileAnalysisSkipped = isIgnoredFileAnalysisSkipped;
-        this.ignoredAuthorsList = ignoredAuthorsList;
-        this.isIgnoredAuthorsListOverriding = isIgnoredAuthorsListOverriding;
-        this.extraOutputFolderName = extraOutputFolderName;
-
-        String organization = location.getOrganization();
-        String repoName = location.getRepoName();
-        displayName = repoName + "[" + branch + "]";
-        outputFolderName = repoName + "_" + branch;
-        repoFolderName = repoName;
-
-        if (!organization.isEmpty()) {
-            repoFolderName = organization + "_" + repoFolderName;
-            displayName = organization + "/" + displayName;
-            outputFolderName = organization + "_" + outputFolderName;
+    /**
+     * Processes the organization name of the repository and updates the relevant
+     * names of the repository configs.
+     */
+    private void processOrganization(RepoBuilder repoBuilder) {
+        String org = location.getOrganization();
+
+        if (!org.isEmpty()) {
+            this.repoFolderName = repoBuilder.repoFolderName == null
+                    ? org + "_" + this.repoFolderName
+                    : repoBuilder.repoFolderName;
+            this.displayName = repoBuilder.displayName == null
+                    ? org + "/" + this.displayName
+                    : repoBuilder.displayName;
+            this.outputFolderName = repoBuilder.outputFolderName == null
+                    ? org + "_" + this.outputFolderName
+                    : repoBuilder.outputFolderName;
+        }
+    }
+
+    /**
+     * Builds the necessary configurations for RepoConfiguration.
+     * Obeys the Builder pattern as described in {@link CliArguments}.
+     */
+    public static class RepoBuilder {
```

Comment:
minor: since CliArguments calls it's builder `Builder`, we could also use RepoConfiguration.Builder for consistency

Reply:
Sure thing, calling it `Builder` would make the code's naming conventions more consistent throughout the codebase! Will work on this ASAP!